Mathlib/Analysis/Normed/Ring/Basic.lean

```diff
@@ -736,7 +730,39 @@ theorem norm_prod (s : Finset β) (f : β → α) : ‖∏ b ∈ s, f b‖ = ∏
 theorem nnnorm_prod (s : Finset β) (f : β → α) : ‖∏ b ∈ s, f b‖₊ = ∏ b ∈ s, ‖f b‖₊ :=
   map_prod nnnormHom.toMonoidHom f s
 
-end NormedCommRing
+end SeminormedCommRing
+
+section NormedAddCommGroup
+variable [NormedAddCommGroup α] [MulOneClass α] [NormMulClass α] [Nontrivial α]
+
+/-- Deduce `NormOneClass` from `NormMulClass` under a suitable nontriviality hypothesis. Not
+an instance, in order to avoid loops with `NormOneClass.nontrivial`. -/
+lemma NormMulClass.toNormOneClass : NormOneClass α where
+  norm_one := by
+    obtain ⟨u, hu⟩ := exists_ne (0 : α)
+    simpa [mul_eq_left₀ (norm_ne_zero_iff.mpr hu)] using (norm_mul u 1).symm
+
+end NormedAddCommGroup
+
+section NormedRing
+
+variable [NormedRing α] [NormMulClass α]
+
+instance NormMulClass.isAbsoluteValue_norm : IsAbsoluteValue (norm : α → ℝ) where
+  abv_nonneg' := norm_nonneg
+  abv_eq_zero' := norm_eq_zero
+  abv_add' := norm_add_le
+  abv_mul' := norm_mul
+
+instance NormMulClass.toNoZeroDivisors : NoZeroDivisors α where
+  eq_zero_or_eq_zero_of_mul_eq_zero {a b} h := by
+    simpa only [← norm_eq_zero (E := α), norm_mul, mul_eq_zero] using h
+
+/-- A non-zero normed ring satisfying `NormMulClass` is a domain. Not an instance, to avoid
+loops with `IsDomain.toNontrivial`. -/
+lemma NormMulClass.toIsDomain [Nontrivial α] : IsDomain α := NoZeroDivisors.to_isDomain α
```

Comment:
I'd suggest just dropping this, the end user can use `NoZeroDivisors.to_isDomain` directly if they want it.